daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -192,8 +219,50 @@ func (l *logStream) Close() error {
 	return nil
 }
 
-// create creates a log stream for the instance of the awslogs logging driver
+// create creates log group and log stream for the instance of the awslogs logging driver
 func (l *logStream) create() error {
+	if err := l.createLogStream(); err != nil {
+		if l.logCreateGroup {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == resourceNotFoundCode {
```

Comment:
Calling `DescribeLogGroups` will require the `logs:DescribeLogGroups` permission in the user's IAM policy.  I'd generally lean toward the current implementation as it will require fewer permissions; the only concern with the current implementation would be the possibility of engaging throttles on `CreateLogGroup` or `CreateLogStream`.  I can reach out to the CloudWatch Logs team if you'd like.